pkg/rds/adapters.go

```diff
@@ -0,0 +1,320 @@
+package rds
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dbtuneai/agent/pkg/agent"
+	guardrails "github.com/dbtuneai/agent/pkg/guardrails"
+	"github.com/dbtuneai/agent/pkg/internal/keywords"
+	"github.com/dbtuneai/agent/pkg/internal/utils"
+	"github.com/dbtuneai/agent/pkg/pg"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type RDSAdapter struct {
+	agent.CommonAgent
+	Config            Config
+	GuardrailSettings guardrails.Config
+	State             State
+	AWSClients        AWSClients
+	PGDriver          *pgxpool.Pool
+}
+
+func CreateRDSAdapter(configKey *string) (*RDSAdapter, error) {
+	var keyValue string
+	if configKey == nil {
+		keyValue = RDS_CONFIG_KEY
+	} else {
+		keyValue = *configKey
+	}
+
+	var err error
+	var config Config
+	config, err = ConfigFromViper(keyValue)
+	if err != nil {
+		return nil, fmt.Errorf("failed to bind RDS config: %w", err)
+	}
```

Comment:
This taking in of an optional `configKey`, i.e. `*configKey`, is because aurora will use this exact same adapter but requires parsing `aurora:` from the yaml config, while using plain RDS requires parsing `rds:`

Reply:
I left them separate in case they do ever need to change, that's why at the bottom of this file there's an `AuroraRDSAdapater`, it just so happens to use the `RDSAdapater`.

If you're refering to the error string, yeah I should fix that!